### PR TITLE
Add a new quick start using Redisson Java Redis client

### PR DESCRIPTION
**Pull request by majguo**
## Purpose

* Add a new quickstart which incorporates Azure Cache for Redis into a Java app using the Redisson Redis client.

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class App {
+    public static void main(String[] args) {
+        // Connect to the Azure Cache for Redis over the TLS/SSL port using the key
+        Config config = new Config();
+        config.useSingleServer().setPassword(System.getenv("REDISCACHEKEY"))
+                .setAddress(String.format("rediss://%s:6380", System.getenv("REDISCACHEHOSTNAME")));
+        RedissonClient client = Redisson.create(config);
+
+        // Perform cache operations using the cache client
+        RMap<String, String> cache = client.getMap("namedCache");
```
Comment (brendanzagaeski) [inline quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java:23]: Since I'm not too familiar with the Redisson library, I'm wondering if the names `cache` and `namedCache` are good matches for the Redisson concept of a map.

From a quick skim of the Redisson docs, my first impression is that they tend to use `map` as the variable name, and the string they pass to `getMap()` doesn't typically include the word "cache" either:

```
RMap<String, String> map = redisson.getMap("test");
```

I'm OK with either the current naming in this PR or naming similar to the Redisson samples, depending on what fits best with how Redisson works. Based on <https://github.com/redisson/redisson/wiki/11.-Redis-commands-mapping>, it looks like `RMap` is specifically for operations where Redis automatically does hashing. It looks like this is the most popular way to use a Redis cache from Redisson, but Redisson technically does support other methods of setting values. It looks like the convention of preferring RMap in Redisson might be inspired by _Use hashes when possible_ as mentioned on <https://redis.io/topics/memory-optimization>.

Long story short, I think maybe borrowing the `map` naming from the Redisson samples will be a good fit?

File: quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        System.out.println("\nCache Command  : SET Message");
+        System.out.println("Cache Response : " + cache.put("Message",
+                String.format("Hello! The cache is working from Java! %s", LocalDateTime.now())));
+
+        // Demonstrate "SET Message" executed as expected
+        System.out.println("\nCache Command  : GET Message");
+        System.out.println("Cache Response : " + cache.get("Message"));
+
+        System.exit(0);
```
Comment (brendanzagaeski) [inline quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java:37]: In https://github.com/redisson/redisson-examples/blob/master/objects-examples/src/main/java/org/redisson/example/objects/SSLExamples.java, I noticed there's a call to `RedissonClient.shutdown()` instead of `System.exit(0)`. Is that appropriate here too?

```suggestion
        client.shutdown();
        System.exit(0);
```

From <https://github.com/redisson/redisson/wiki/16.-FAQ#q-when-do-i-need-to-shut-down-a-redisson-instance-at-the-end-of-each-request-or-the-end-of-the-life-of-a-thread>, it sounds like calling `shutdown()` isn't necessary when the app is exiting anyway, so probably this isn't necessary. Does that sound right?

File: quickstart/java-redisson/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+## Run the sample
+
+[Download the sample code to your development PC.](/README.md#get-the-samples)
+
+Change directories to the folder containing this sample.
+
+Execute the following Maven command to build and run the app:
+
+```CMD
+mvn compile exec:java -Dexec.mainClass=example.demo.App
```
Comment (brendanzagaeski) [inline quickstart/java-redisson/README.md:49]: I suspect this might be due to the particular details of the Maven packing, so it might not be worth trying to fix for this sample, but I noticed that when I execute the sample on the command line, before the program output itself, I get:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

Do you know if there's any easy way to fix or silence those messages so new users won't need to worry about them when they try running this sample?

**Review by brendanzagaeski — COMMENTED** [review]
Thanks so much for creating this sample! This looks great. I'll left a few tiny ideas and questions that I think will be quick to address. After that, I think this will be ready to merge.

File: quickstart/java-redisson/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (majguo) [reply quickstart/java-redisson/README.md]: Good idea to make it be compatible with PowerShell, suggested changes are committed, thanks.

File: quickstart/java-redisson/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- java
+name: 'Quickstart: Use Azure Cache for Redis in Java with Redisson Redis client'
+description: Learn how to incorporate Azure Cache for Redis into a Java app using the Redisson Redis client.
+products:
+- azure
+- azure-cache-redis
+---
+# Quickstart: Use Azure Cache for Redis in Java with Redisson Redis client
+
+This sample show you how to incorporate Azure Cache for Redis into a Java app using the [Redisson](https://redisson.org/) Redis client. See the [accompanying article](https://aka.ms/azure-cache-java-redisson-quickstart) on the documentation site for details, including best practices and how to create the sample code from scratch.
```
[the remark this answers is not in the document]
Comment (majguo) [reply quickstart/java-redisson/README.md:13]: Sounds good, suggested changes are committed, thanks.

File: quickstart/java-redisson/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+Execute the following Maven command to build and run the app:
+
+```CMD
+mvn compile exec:java -Dexec.mainClass=example.demo.App
+```
+
+## References
+
+* [Quickstart article on the documentation site](https://aka.ms/azure-cache-java-redisson-quickstart)
```
[the remark this answers is not in the document]
Comment (majguo) [reply quickstart/java-redisson/README.md:54]: Sounds good, done.

File: quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java (hunk 1 above)
Comment (majguo) [reply quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java]: Good catch again! Updated the variable naming in commit 5e6be51.

File: quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java (hunk 2 above)
Comment (majguo) [reply quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java:37]: I tested the following cases:
* Just removing `System.exit(0);` and it turns out that the process won't exit;
* Call `client.shutdown();` at the end will successfully exit the process.

So I prefer to your suggested changes and committed.

Comment (majguo) [reply quickstart/java-redisson/redis-redisson-test/src/main/java/example/demo/App.java]: BTW, once `client.shutdown();` is called, the process will successfully exit, so `System.exit(0);` is no longer needed. Pushed commit 7f72e29 to get it removed.

File: quickstart/java-redisson/README.md (hunk 3 above)
Comment (majguo) [reply quickstart/java-redisson/README.md]: Good catch. I googled a [solution](https://stackoverflow.com/questions/7421612/slf4j-failed-to-load-class-org-slf4j-impl-staticloggerbinder) and pushed commit 2a43397 to resolve warnings.

**Review by brendanzagaeski — APPROVED** [review]
